### PR TITLE
 Reactive MS SQL client support

### PR DESCRIPTION
**Pull request by fedinskiy**
Reactive MS SQL client support as tech preview. Part of QUARKUS-1081

Commits will be squashed before or during the merge

### CODE REVIEW

File: sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/dbpool/PostgresPoolIT.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 @QuarkusTest
 @TestProfile(PostgresqlTestProfile.class)
 @TestMethodOrder(OrderAnnotation.class)
-public class PostgresPoolTest {
+public class PostgresPoolIT {
```
Comment (Sgitario) [inline sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/dbpool/PostgresPoolIT.java:39]: This test should be executed by surefire, so this change needs to be reverted.

File: sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/Db2HandlerIT.java [hunk 2]
```diff
@@ -11,18 +11,24 @@
     private static final String DATABASE = "amadeus";
 
     @Container(image = "${db2.image}", port = 50000, expectedLog = "Setup has completed")
-    static Db2Service db2 = new Db2Service()
-            .with("test", "test", DATABASE);
+    static DefaultService db2 = new DefaultService()
```
Comment (Sgitario) [inline sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/Db2HandlerIT.java:14]: It should be Db2Service.

File: sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/MSsqlHandlerIT.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class MSsqlHandlerIT extends CommonTestCases {
+    private static final String MSSQL_PASSWORD = "QuArKuS_tEsT";
+    private static final String DATABASE = "msdb";
+    private static final int MSSQL_PORT = 1433;
+
+    @Container(image = "${mssql.image}", port = MSSQL_PORT, expectedLog = "Service Broker manager has started")
+    static DefaultService mssql = new DefaultService()
```
Comment (Sgitario) [inline sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/MSsqlHandlerIT.java:16]: It should be SqlServerService

**Review by Sgitario — CHANGES_REQUESTED** [review]
Also, rebase your PR.

File: sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/MSsqlHandlerIT.java (hunk 3 above)
Comment (fedinskiy) [reply sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/MSsqlHandlerIT.java:16]: Why? At the moment SqlService has no onPreStart action, so all properties should be configured manually anyway, due to errors in constant values and in generating code getXUrl methods couldn't be used either. I'll fix it in framework, but until 0.0.10 there is no point in using it here.

Comment (Sgitario) [reply sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/MSsqlHandlerIT.java:16]: I was not aware of this issue. Please, submit an issue in the test framework repository and add a link here to state that will be fixed in 0.0.10 and we should change it to use SqlServerService.
Moreover, apply your workaround in the OpenShiftMssqlDatabaseIT test too.

**Review by Sgitario — CHANGES_REQUESTED** [review]
Just one minor comment to add ++ please rebase your BR, at the moment, it cannot be merged because there are conflicts.

File: sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/MSsqlHandlerIT.java (hunk 3 above)
Comment (fedinskiy) [reply sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/MSsqlHandlerIT.java:17]: Done and done https://github.com/quarkus-qe/quarkus-test-framework/issues/251

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
OCP are not working due to:
- We're running out of space
- The Jenkins nodes is not configured to use the RH registry
Merging

**Comment by Sgitario** [comment]
@fedinskiy please backport it to 2.2

**Pull request merged**